src/main/java/com/amazonaws/mobileconnectors/iot/AWSIotMqttManager.java

```diff
@@ -710,18 +710,23 @@ public void connect(final AWSIotMqttClientStatusCallback statusCallback) {
         // create a thread as the credentials provider getCredentials() call may require
         // a network call and will possibly block this connect() call
         new Thread(new Runnable() {
-            @Override
-            public void run() {
-                if (mqttLWT != null) {
-                    options.setWill(mqttLWT.getTopic(), mqttLWT.getMessage().getBytes(),
-                            mqttLWT.getQos().asInt(), false);
-                }
-                AWSIotWebSocketUrlSigner signer = new AWSIotWebSocketUrlSigner("iotdata");
-                String mqttWebSocketURL = signer.getSignedUrl(webSocketEndpoint,
-                        clientCredentialsProvider.getCredentials(), System.currentTimeMillis());
-                options.setServerURIs(new String[] {mqttWebSocketURL});
-                mqttConnect(options, statusCallback);
-            }
+			@Override
+			public void run() {
+				try {
+					if (mqttLWT != null) {
+						options.setWill(mqttLWT.getTopic(), mqttLWT.getMessage().getBytes(), mqttLWT.getQos().asInt(),
+								false);
+					}
+					AWSIotWebSocketUrlSigner signer = new AWSIotWebSocketUrlSigner("iotdata");
+					String mqttWebSocketURL = signer.getSignedUrl(webSocketEndpoint,
+							clientCredentialsProvider.getCredentials(), System.currentTimeMillis());
+					options.setServerURIs(new String[] { mqttWebSocketURL });
+					mqttConnect(options, statusCallback);
+				} catch (java.lang.IllegalStateException e) {
```

Comment:
why qualify java.lang?

Reply:
I am shying way from catching all exceptions, since other exceptions that really need to be brought to our attention should fail fast.  In this case, the exception that we need to catch is java.lang.IllegalStateException, since this is thrown when we modify an option object that has been locked as immutable.

It also makes sense to catch all exceptions here as we will just retry a connect later, it is safe to avoid crashes and us having to add another exception type.  Do you prefer catch all?

Reply:
I see, IllegalStateException would have been sufficient with a package import.  No particular really good reason.  I can try to justify it saying that catching IllegalStateException, should be rare, so I want to highlight it in the code that we are doing this by qualifying with java.lang.IllegalStateException.  If this were a regular type of exception (checked file exception, for example), I would not add java.lang it.